### ISSUE TITLE
Duplicate Videos

### ISSUE THREAD

**Issue by Alterak** [Chocobozzz/PeerTube#3008]
Some local videos are displayed 2 times on the page. I checked the uploaded videos several times, these videos are uploaded 1 time. And also the directory ID on the server is available only once.
Can you help solve this problem?

https://peervideo.ru
Peertube version 2.3.0
Browser - Firefox 78.0.2 (x64)

![double](https://user-images.githubusercontent.com/11453567/88534238-5148c880-d010-11ea-9197-26fffa781ace.PNG)

Log Browser

> Reusing component trending-videos-list{"a-state":"42"}. main-es2015.0d10dbd20f82401ddd8f.js:3:1410676
> Ошибка карты кода: Error: NetworkError when attempting to fetch resource.
> URL ресурса: moz-extension://dd019ae5-49ad-447d-8a9b-0d147c0a7fdf/content/js/purify.min.js
> URL карты кода: purify.min.js.map
> 
> Injecting 0 themes. main-es2015.0d10dbd20f82401ddd8f.js:3:1639138
> Client credentials loaded. main-es2015.0d10dbd20f82401ddd8f.js:3:1553950
> Enabling default theme. main-es2015.0d10dbd20f82401ddd8f.js:3:1640252
> Checking if the initial state has a scroll. main-es2015.0d10dbd20f82401ddd8f.js:3:1547211
> Firefox не может установить соединение с сервером wss://peervideo.ru/socket.io/?accessToken=e394673786e8e69d35551685865882b1ec3ba27f&EIO=3&transport=websocket&sid=xkzdMyIU9YgDW18zAACC.

**Comment by Alterak** [comment]
Sometimes there are no duplicate videos, but sometimes they appear. I also noticed double videos from remote subscribers' servers.

**Comment by Chocobozzz** [comment]
Maybe the order changed after some views on videos, and so were moved from page 1 to page 2. Then when you load page 2 you reload videos that were moved in the ranking

**Comment by Alterak** [comment]
> Maybe the order changed after some views on videos, and so were moved from page 1 to page 2. Then when you load page 2 you reload videos that were moved in the ranking

Duplicate videos appear randomly, different videos after page refresh. Now, for example, these videos are being duplicated

![1](https://user-images.githubusercontent.com/11453567/89064693-22fe1c80-d373-11ea-98d2-14fd3caf6fec.PNG)
![2](https://user-images.githubusercontent.com/11453567/89064701-25607680-d373-11ea-86ab-4cc6b835acba.PNG)

**Comment by Chocobozzz** [comment]
> There are duplicates of videos that were downloaded earlier, before the disc was accidentally deleted. And also duplicated videos from subscribers that were before the disc was deleted. New videos are NOT duplicated!

Then it's related to your database corruption/reimport, and not really a PeerTube bug.
